src/fmu/dataio/_model/enums.py

```diff
@@ -56,6 +56,20 @@ def _missing_(cls: Type[Content], value: object) -> None:
         )
 
 
+class ErtSimulationMode(str, Enum):
+    """The simulation mode ert was run in. These definitions come from
+    `ert.mode_definitions`."""
+
+    ensemble_experiment = "ensemble_experiment"
+    ensemble_smoother = "ensemble_smoother"
+    es_mda = "es_mda"
```

Comment:
More a question, not only for the new code but for all in this module: Why is type hints not applied?

```
ensemble_experiment: str = "ensemble_experiment"
...
```

Reply:
I've never thought about this for enums and I'm not sure. I think basically because the enum itself _becomes_ the type, rather than a container containing types. And since we have the str mixin `class ErtSimulationMode(str, Enum)` mypy already know that every value should be a string.

Python 3.11 has `StrEnum` which enforces the type at runtime, a better option for the future